test/utils.spec.js

```diff
@@ -262,15 +285,31 @@ describe('utils', function() {
 });
 
 /* eslint-disable no-console */
-function handleConsoleWarns(doWithShimmedConsoleWarn, onWarn) {
-  // Shim `console.warn`
-  var _warn = console.warn;
+function handleWarnings(doWithShimmedConsoleWarn, onWarn) {
+  /* eslint-disable no-console */
+  if (typeof process.emitWarning !== 'function') {
+    // Shim `console.warn`
+    var _warn = console.warn;
+    console.warn = onWarn;
 
-  console.warn = onWarn;
+    doWithShimmedConsoleWarn();
 
-  doWithShimmedConsoleWarn();
+    // Un-shim `console.warn`,
+    console.warn = _warn;
 
-  // Un-shim `console.warn`
-  console.warn = _warn;
+    /* eslint-enable no-console */
+  } else {
+    /* eslint-disable no-inner-declarations */
```

Comment:
And here, can we either re-enable this lint after the inner declaration or use `eslint-disable-line`?